examples/vag_cdc.c

```diff
@@ -0,0 +1,260 @@
+/*
+ * VAG_CDC.c
+ *
+ * Created: 23.06.2013 20:00:51
+ *  Author: Dennis Schuett, dev.shyd.de
+ */
+
+#define F_CPU 8000000UL
+
+#include <inttypes.h>
+#include <avr/io.h>
+#include <avr/interrupt.h>
+#include <util/delay.h>
+
+#include "uart.h"
+
+#define UART_BAUD_RATE 9600
+#define LED_PWR PA0
+#define RADIO_OUT PD2
+#define FT_CBUS1 PD3
+#define RADIO_OUT_IS_HIGH (PIND & (1<<RADIO_OUT))
```

Comment:
Let's break this down. 

---

`RADIO_OUT` is the `PD2` register, so we can resolve this into two operations.

```c
1 << PD2
```

The first is a [bitwise left-shift](https://en.wikipedia.org/wiki/Bitwise_operations_in_C#Left_shift_%3C%3C). We shift `1` to the left by the _n_ bits, where _n_ is the current value of the `PD2` register. This has the effect of providing a `1` bit at an arbitrary location, e.g. `1<<0 == 0b0000_0001`, `1<<1=0b0000_0010`, `1<<2=0b0000_0100`. This is basically saying "set bit _n_ to 1."

---

```c
PIND & result
```

The next operation is a [bitwise AND](https://en.wikipedia.org/wiki/Bitwise_operations_in_C#Bitwise_AND_&) between the value of `PIND` and the result of the previous operation. This is essentially saying "check if bit _n_ in `PIND` is set to `1`."

In summary:

> Tell me if the bit number defined by the current value of `PD2` within `PIND` is `1`

In pseudocode, treating it as a list of integers in a bitset (e.g. `PIND=[0, 1, 0, 1, 0, 0, 0, 1]`), each equal to `0` or `1`, instead of a single integer:

```python
PIND[PD2] == 1
```

The result of this operation will inform us as to whether the radio output (presumably `MISO` in SPI for receiving data from the head unit) is set to high, presumably indicating whether